### ISSUE TITLE
Add support for a logo for a template

### ISSUE THREAD

**Issue by dselman** [accordproject/cicero#240]
It would be very useful for display purposes if a Cicero template could contain an **optional** logo.

- File is called: `logo.png` and is in the root of the template
- Must be a PNG file
- Maximum size is 256x256 pixels

**Comment by gagandeepp** [comment]
@dselman  @mttrbrts  is this still open,I can have a look into this?

**Comment by mttrbrts** [comment]
Thanks @gagandeepp, yes, I'll release the assignment!

**Comment by gagandeepp** [comment]
how to test this functionality??

**Comment by gagandeepp** [comment]
Any updates?

**Comment by irmerk** [comment]
@mttrbrts pinging you again on this one.

**Comment by algomaster99** [comment]
@jeromesimeon I would like to give this a try. :)

**Comment by jeromesimeon** [comment]
There hasn't been any progress on this. It's yours @algomaster99  !

**Comment by jeromesimeon** [comment]
What should the proper internal representation for an image in JavaScript?

I am suspecting it shouldn't be a string. Should it be instead:
- a buffer object?
- a `Blob` object?  
- an `Image` object? https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Using_images

If we need manipulation of the image (or even simply know the image size) do we need support for a library. e.g., https://github.com/lovell/sharp ?

**Comment by jeromesimeon** [comment]
Another open question: should the logo file be part of the hash calculated for the template?

**Comment by jeromesimeon** [comment]
@Michael-Grover is there a logo for Accord Project templates that you would like? (Probably should be added to the templates in the library once the  functionality is in place).

Would the `256x256` proposed size be right (e.g., if we want to use it on the template library Web site somewhere, or in the template library component in Cicero UI)?

**Comment by algomaster99** [comment]
@jeromesimeon 
> Another open question: should the logo file be part of the hash calculated for the template?

Can you tell the point of calculating hash? :sweat_smile:

**Comment by jeromesimeon** [comment]
> @jeromesimeon
> 
> > Another open question: should the logo file be part of the hash calculated for the template?
> 
> Can you tell me why do we calculate hash? 😅

It's a way to uniquely identify the content of the template (can be used to do sanity checks on downloads for example).

**Comment by Michael-Grover** [comment]
> @Michael-Grover is there a logo for Accord Project templates that you would like? (Probably should be added to the templates in the library once the functionality is in place).
> 
> Would the `256x256` proposed size be right (e.g., if we want to use it on the template library Web site somewhere, or in the template library component in Cicero UI)?

![Accord](https://user-images.githubusercontent.com/40569793/77349787-eb6ecc00-6d11-11ea-8516-81bc7ff5bdc9.png)
@jeromesimeon how's this?

256x256 seems really big. What applications might we use for the logo that are 256 x 256, or 128 x 128 on a retina screen?

**Comment by jeromesimeon** [comment]
Summer of some fact-finding for png files size.

How to determine that the file is PNG:
- Specification: http://www.libpng.org/pub/png/spec/1.2/PNG-Rationale.html#R.PNG-file-signature
- Sample code: https://github.com/arian/pngjs/blob/c1a2f5852686a22229f5f3154f974011162b5263/PNGReader.js#L80

How to determine the size:
- `image-size` support "fried png" but I don't think we have to. See: http://www.jongware.com/pngdefry.html
- If we don't this big of code just seems to work:
```
const fs = require('fs');
const logoBuffer = fs.readFileSync('logo.png');

function calculate(buffer) {
    return {
        height: buffer.readUInt32BE(20),
        width: buffer.readUInt32BE(16)
    }
}

console.log(JSON.stringify(calculate(logoBuffer)));
```

One this logo:
![logo](https://user-images.githubusercontent.com/670099/77772350-5b38cb80-701e-11ea-8131-8050165445e2.png)

Trace:
```
bash-3.2$ node test.js 
{"height":128,"width":128}
```

**Comment by jeromesimeon** [comment]
Fixed by @algomaster99 and merged in #533 with some additional fixes in #535
